src/screens/home/HomeScreen.tsx

```diff
@@ -183,15 +184,44 @@ const ExpandedContent = (bottomSheetBehavior: BottomSheetBehavior) => {
 
 export const HomeScreen = () => {
   const navigation = useNavigation();
-  const {userStopped} = useStorage();
+  const {userStopped, setCheckInJSON} = useStorage();
   useEffect(() => {
     if (__DEV__ && TEST_MODE) {
       DevSettings.addMenuItem('Show Demo Menu', () => {
         navigation.navigate('TestScreen');
       });
     }
   }, [navigation]);
+  useEffect(() => {
+    function handleOpenURL(url: any) {
+      const objConstructor = {}.constructor;
+      let urlObj = url;
+      if (url.constructor === objConstructor) {
+        urlObj = url.url;
+      }
+      const routeName = urlObj.split('/')[2];
```

Comment:
Let's break this functionality out into a utility function to make it easier to test / re-use and update across screens